src/App.tsx

```diff
@@ -1,61 +1,129 @@
 import './App.scss';
+import { ChangeEvent, useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+function getUserById(id: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === id);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [userName, setUserName] = useState(0);
+  const [title, setTitle] = useState('');
+  const [visibleTodos, setTodos] = useState(todos);
+  const [isErrorTitle, setErrorTitle] = useState(false);
+  const [isErrorUserName, setErrorUserName] = useState(false);
+
+  const handleChangeUser = (event: ChangeEvent<HTMLSelectElement>) => {
+    setUserName(Number(event.target.value));
+    setErrorUserName(false);
+  };
+
+  const handleTitle = ((event: ChangeEvent<HTMLInputElement>) => {
+    const input = event.target.value;
+    const filteredInput = input.replace(/[^A-Za-zA-Яa-я0-9\s]/g, '');
+
+    setTitle(/^\s*$/.test(filteredInput)
```

Comment:
move this to the variable
```suggestion
    setTitle(...)
```

Reply:
just remove this reGex, bc i can't write a space